src/panels/lovelace/editor/card-editor/hui-dialog-create-card.ts

```diff
@@ -0,0 +1,248 @@
+                "ui.panel.lovelace.editor.cardpicker.by_card"
+              )}
+            ></mwc-tab>
+            <mwc-tab
+              .label=${this.hass!.localize(
+                "ui.panel.lovelace.editor.cardpicker.by_entity"
+              )}
+            ></mwc-tab>
+          </mwc-tab-bar>
+        </div>
+        ${this._currTabIndex === 0
+          ? html`
+              <hui-card-picker
+                .lovelace=${this._params.lovelaceConfig}
+                .hass=${this.hass}
+                @config-changed=${this._handleCardPicked}
+              ></hui-card-picker>
+            `
+          : html`
+              <hui-entity-picker
+                .hass=${this.hass}
+                @selected-changed=${this._handleSelectedChanged}
+              ></hui-entity-picker>
+            `}
+
+        <div slot="primaryAction">
+          <mwc-button @click=${this._cancel}>
+            ${this.hass!.localize("ui.common.cancel")}
+          </mwc-button>
+          ${this._currTabIndex === 1
+            ? html`
+                <mwc-button @click=${this._suggestCards}>
+                  ${this.hass!.localize("ui.common.continue")}
+                </mwc-button>
+              `
+            : ""}
+        </div>
+      </ha-dialog>
+    `;
+  }
+
+  private _ignoreKeydown(ev: KeyboardEvent) {
+    ev.stopPropagation();
+  }
+
+  static get styles(): CSSResultArray {
+    return [
+      haStyleDialog,
+      css`
+        @media all and (max-width: 450px), all and (max-height: 500px) {
+          /* overrule the ha-style-dialog max-height on small screens */
+          ha-dialog {
+            --mdc-dialog-max-height: 100%;
+            height: 100%;
+          }
+        }
+
+        @media all and (min-width: 850px) {
+          ha-dialog {
+            --mdc-dialog-min-width: 845px;
+          }
+        }
+
+        ha-dialog {
+          --mdc-dialog-max-width: 845px;
+        }
+
+        ha-header-bar {
+          --mdc-theme-on-primary: var(--primary-text-color);
+          --mdc-theme-primary: var(--mdc-theme-surface);
+          flex-shrink: 0;
+          border-bottom: 1px solid
+            var(--mdc-dialog-scroll-divider-color, rgba(0, 0, 0, 0.12));
+        }
+
+        @media (min-width: 1200px) {
+          ha-dialog {
+            --mdc-dialog-max-width: calc(100% - 32px);
+            --mdc-dialog-min-width: 1000px;
+          }
+        }
+
+        .header_button {
+          color: inherit;
+          text-decoration: none;
+        }
+
+        mwc-tab-bar {
+          padding-top: 8px;
+        }
+      `,
+    ];
+  }
+
+  private _handleCardPicked(ev) {
+    const config = ev.detail.config;
+    if (this._params!.entities && this._params!.entities.length) {
+      if (Object.keys(config).includes("entities")) {
+        config.entities = this._params!.entities;
+      } else if (Object.keys(config).includes("entity")) {
+        config.entity = this._params!.entities[0];
+      }
+    }
```

Comment:
Isn't this logic in the edit card dialog? As it is also used for the suggest card?

Reply:
I dont know what you mean by this

Reply:
You can just add `entities` to the options of `showEditCardDialog`.

Reply:
Oh wait... `create-card-dialog` will open `suggest-card-dialog` and that will open `create-card-dialog` again and this logic kicks in?

Reply:
But I don't see that in the code now, suggest card will open `dialog-edit-card`? Does it even work now?